detection_rules/packaging.py

```diff
@@ -147,18 +84,14 @@ def __init__(self, rules: List[TOMLRule], name: str, deprecated_rules: Optional[
         self.release = release
         self.registry_data = registry_data or {}
 
-        self.changed_rule_ids, self.new_rules_ids, self.removed_rule_ids = self._add_versions(current_versions,
-                                                                                              update_version_lock,
-                                                                                              verbose=verbose)
+        if min_version is not None:
+            self.rules = [r for r in self.rules if min_version <= r.contents.latest_version]
 
-        if min_version or max_version:
-            self.rules = [r for r in self.rules
-                          if (min_version or 0) <= r.contents['version'] <= (max_version or r.contents['version'])]
+        if max_version is not None:
+            self.rules = [r for r in self.rules if max_version >= r.contents.latest_version]
 
-    def _add_versions(self, current_versions, update_versions_lock=False, verbose=True):
-        """Add versions to rules at load time."""
-        return manage_versions(self.rules, current_versions=current_versions,
-                               save_changes=update_versions_lock, verbose=verbose)
+        self.changed_ids, self.new_ids, self.removed_ids = \
+            manage_versions(self.rules, verbose=False, save_changes=False)
```

Comment:
```suggestion
            manage_versions(self.rules, verbose=verbose, save_changes=False)
```

Reply:
sure, we can do this. these flags are pretty messy and hard to trace. but yeah i can set this back